sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -2399,11 +2405,12 @@ constexpr void check_lsc_block_2d_restrictions() {
           "Unsupported number of blocks for 2D load/prefetch");
       static_assert(BlockHeight <= 32, "Unsupported block height for load");
     }
-    static_assert(BlockWidth * sizeof(T) >= 4 &&
-                      BlockWidth * NBlocks * sizeof(T) <= 64,
-                  "Unsupported block width");
+    static_assert(BlockWidth * sizeof(T) >= 4, "Unsupported block width");
+    __ESIMD_BLOCK_2D_WIDTH_CHECK(Op, BlockWidth, NBlocks, sizeof(T));
```

Comment:
Sorry, what do we get by moving this to a macro? Do we plan to reuse it elsewhere?

Reply:
Yes.